spec/railway_ipc/consumer_spec.rb

```diff
@@ -22,40 +15,287 @@
     end
   end
 
-
   it "registers the handler for the message" do
     RailwayIpc::Consumer.handle(LearnIpc::Commands::TestMessage, with: RailwayIpc::TestHandler)
     expect(RailwayIpc::ConsumerResponseHandlers.instance.registered)
         .to include("LearnIpc::Commands::TestMessage")
   end
 
-  it "routes the message to the correct handler" do
-    allow(RailwayIpc::TestHandler).to receive(:new).and_return(handler_instance)
-    expect(handler_instance).to receive(:handle).with(instance_of(LearnIpc::Commands::TestMessage))
-    consumer.work(payload)
-  end
+  describe '#work_with_params' do
+    let(:consumer) { RailwayIpc::TestConsumer.new }
 
-  it "acks the message" do
-    allow(RailwayIpc::TestHandler).to receive(:new).and_return(handler_instance)
-    expect(handler_instance).to receive(:ack!)
-    consumer.work(payload)
-  end
+    context 'when message is successfully decoded with known message type' do
+      context 'when a consumed message records alreadys exists' do
+        context 'when message has a status of "success"' do
+          let!(:test_message) { test_message_stub }
+          let!(:payload) { payload_stub(message: test_message) }
+          let!(:delivery_info) { delivery_info_stub }
+          let!(:consumed_message) { consumed_message_stub(message: test_message, status: RailwayIpc::ConsumedMessage::STATUSES[:success]) }
+          let!(:test_handler) { RailwayIpc::TestHandler.new }
+
+          it 'does not update the record' do
+            expect {
+              consumer.work_with_params(payload, delivery_info, nil)
+            }.to_not change { consumed_message.updated_at }
+          end
+
+          it 'does not process the message' do
+            allow(RailwayIpc::TestHandler).to receive(:new).and_return(test_handler)
+            expect(test_handler).not_to receive(:handle)
+
+            consumer.work_with_params(payload, delivery_info, nil)
+          end
+
+          it 'acks the message' do
+            allow(RailwayIpc::TestHandler).to receive(:new).and_return(test_handler)
+            expect(test_handler).to receive(:ack!)
+
+            consumer.work_with_params(payload, delivery_info, nil)
+          end
+        end
+
+        context 'when message has status of "unknown_message_type"' do
```

Comment:
So this is basically saying. 

1. If a the payload is decoded with a known message
2. The message in the payload alreadys has a matching record in the db with the same uuid 
3. And the message has a type of :unkown_message_type

It should not process it, update the record, but it should acknowledge it was received.

But if I'm understanding your question, we should check to see if we are now handling that message. Which should be taking place here: https://github.com/learn-co/railway_ipc_gem/pull/9/files#diff-d2b80ef0d2ea8f9b6b718a9a4f00e780R31-R35

and then reprocessed with a new status here: https://github.com/learn-co/railway_ipc_gem/pull/9/files#diff-d2b80ef0d2ea8f9b6b718a9a4f00e780R73

I think that is a test case I did miss though. Thanks for raising. I'll add an additional test for that to just be specific.

Reply:
If we now know what the message is though, wouldn't we want to update the consumed msg in the db? Otherwise that re-processing will show a message with a type unkonwn that has been processed? That seems a bit confusing. 
But you're totally right, it will be re-processed. Thanks for pointing that out :)

Reply:
Discussed IRL, going to resolve this for now.